indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -160,46 +156,33 @@ private static String makeGroupId(boolean isAppendToExisting, String dataSource)
     }
   }
 
-  @JsonIgnore
   private final IndexIngestionSpec ingestionSchema;
 
-  @JsonIgnore
   private IngestionState ingestionState;
 
-  @JsonIgnore
-  private final AuthorizerMapper authorizerMapper;
-
-  @JsonIgnore
-  private final Optional<ChatHandlerProvider> chatHandlerProvider;
+  private final CircularBuffer<Throwable> buildSegmentsSavedParseExceptions;
 
-  @JsonIgnore
-  private final RowIngestionMeters determinePartitionsMeters;
+  private final CircularBuffer<Throwable> determinePartitionsSavedParseExceptions;
 
-  @JsonIgnore
-  private final RowIngestionMeters buildSegmentsMeters;
+  @MonotonicNonNull
```

Comment:
Could this be `javax.annotation.Nonnull` instead? That seems more common in the codebase, and these objects don't really seem "monotonic"

Reply:
I think `Nonnull` and `MonotonicNonNull` are different. The [javadoc of `Nonnull` says "The annotated element must not be null. Annotated fields must not be null after construction has completed."](https://www.javadoc.io/doc/com.google.code.findbugs/jsr305/latest/javax/annotation/Nonnull.html), while [the one of `MonotonicNonNull` says "Indicates that once the field (or variable) becomes non-null, it never becomes null again."](https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/MonotonicNonNull.html). For the variables annotated as `MonotonicNonNull` in this class, they are nulls when the class is constructed and initialized later in `runTask()`. Once they are initialized, they never become nulls again until the task is finished. So, I think `MonotonicNonNull` is more proper for these variables.

Reply:
Ah, I see, that's good then